lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/variance/test/test.native.js

```diff
@@ -0,0 +1,128 @@
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var resolve = require( 'path' ).resolve;
+var tape = require( 'tape' );
+var tryRequire = require( '@stdlib/utils/try-require' );
+var isNumber = require( '@stdlib/assert/is-number' ).isPrimitive;
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var randu = require( '@stdlib/random/base/randu' );
+var PINF = require( '@stdlib/constants/float64/pinf' );
+var NINF = require( '@stdlib/constants/float64/ninf' );
+var EPS = require( '@stdlib/constants/float64/eps' );
+
+
+// VARIABLES //
+
+var variance = tryRequire( resolve( __dirname, './../lib/native.js' ) );
+var opts = {
+	'skip': ( variance instanceof Error )
+};
+
+
+// TESTS //
+
+tape( 'main export is a function', opts, function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof variance, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'if provided `NaN` for any parameter, the function returns `NaN`', opts, function test( t ) {
+	var v = variance( NaN, 0.5 );
+	t.equal( isnan( v ), true, 'returns NaN' );
+
+	v = variance( 10.0, NaN );
+	t.equal( isnan( v ), true, 'returns NaN' );
+
+	t.end();
+});
+
+tape( 'if provided a nonpositive `a`, the function returns `NaN`', opts, function test( t ) {
+	var y;
+
+	y = variance( 0.0, 2.0 );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = variance( -1.0, 2.0 );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = variance( -1.0, 2.0 );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = variance( NINF, 1.0 );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = variance( NINF, PINF );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = variance( NINF, NINF );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = variance( NINF, NaN );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	t.end();
+});
+
+tape( 'if provided a nonpositive `b`, the function returns `NaN`', opts, function test( t ) {
+	var y;
+
+	y = variance( 2.0, 0.0 );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = variance( 2.0, -1.0 );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = variance( 2.0, -1/0 );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = variance( 1.0, NINF );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = variance( PINF, NINF );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = variance( NINF, NINF );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = variance( NaN, NINF );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	t.end();
+});
+
+tape( 'the function returns the variance of a Kumaraswamy\'s double bounded distribution', opts, function test( t ) {
+	var a;
+	var b;
+	var i;
+	var y;
+
+	// TODO: Add fixtures
```

Comment:
It would be good if we could add test fixtures, but not strictly required as part of this PR.